test/foundry/FrendLend.t.sol

```diff
@@ -0,0 +1,350 @@
+
+        vm.prank(offer.creditor);
+        frendLend.rejectLoanOffer(offerId);
+
+        vm.prank(offer.creditor);
+        vm.expectRevert(FrendLend.NOT_CREDITOR_OR_DEBTOR.selector);
+        frendLend.rejectLoanOffer(offerId);
+    }
+
+    function testNonCreditorOrDebtorCannotReject(address _sender) public {
+        FrendLend.LoanOffer memory offer = _getParams();
+        vm.assume(_sender != offer.creditor && _sender != offer.debtor);
+
+        vm.prank(offer.creditor);
+        uint256 offerId = frendLend.offerLoan{ value: fee }(offer);
+
+        vm.prank(_sender);
+        vm.expectRevert(FrendLend.NOT_CREDITOR_OR_DEBTOR.selector);
+        frendLend.rejectLoanOffer(offerId);
+    }
+
+    ///
+    ////// acceptLoan //////
+    ///
+
+    /**
+        @notice SPEC:
+            Allows a debtor to accept a loan offer, and receive payment
+            This function will:
+                RES1. Delete the offer from storage
+                RES2. Creates a new claim for the loan amount + interest
+                RES3. Transfers the offered loan amount to the debtor
+                RES4. Puts the claim into a non-rejectable repaying state by paying 1 wei
+                RES5. Emits a BullaTagUpdated event with the claimId, the debtor address, a tag, and the current timestamp
+                RES6. Emits a LoanOfferAccepted event with the offerId, the accepted claimId, and the current timestamp
+            Given the following:
+                P1. the current msg.sender is the debtor listed on the offer (covers: offer exists)
+    */
+
+    // SPEC.RES1-7
+    function testAcceptLoan() public {
+        FrendLend.LoanOffer memory offer = _getParams();
+
+        vm.startPrank(offer.creditor);
+        bullaToken.approve(address(frendLend), offer.loanAmount + 1);
+        uint256 offerId = frendLend.offerLoan{ value: fee }(offer);
+        vm.stopPrank();
+
+        uint256 expectedClaimId = BullaClaimERC721(address(bullaClaim)).nextClaimId();
+
+        // RES5, RES6
+        vm.expectEmit(true, true, true, true);
+        emit BullaTagUpdated(bullaClaim.bullaManager(), expectedClaimId, offer.debtor, bytes32(0), block.timestamp);
+        vm.expectEmit(true, true, true, true);
+        emit LoanOfferAccepted(offerId, expectedClaimId, block.timestamp);
+
+        uint256 debtorBalanceBefore = bullaToken.balanceOf(offer.debtor);
+        uint256 creditorBalanceBefore = bullaToken.balanceOf(offer.creditor);
+
+        vm.prank(offer.debtor);
+        frendLend.acceptLoan(offerId, 'TESTURI', bytes32(0));
+
+        // RES1
+        _assertOfferDeleted(offerId);
+
+        // RES2, RES4
+        {
+            Claim memory actualClaim = bullaClaim.getClaim(expectedClaimId);
+            address creditor = ERC721(address(bullaClaim)).ownerOf(expectedClaimId);
+            uint128 claimAmount = uint128(((offer.loanAmount * offer.interestBPS) / MAX_BPS) + offer.loanAmount) + 1;
```

Comment:
Here you are copying the actual implementation in your test. Meaning you are not really testing anything.

Reply:
Try to go from the actualClaimAmount and obtain the loanAmount to see if it matches the offer. Subtle difference, but this way, the implementation code is not used.